services/autorust/codegen/src/codegen_models.rs

```diff
@@ -703,40 +708,65 @@ fn create_struct(cg: &CodeGen, schema: &SchemaGen, struct_name: &str, pageable:
     Ok(code)
 }
 
-struct TypeCode {
-    type_name: TokenStream,
-    code: TokenStream,
+struct FieldCode {
+    type_name: TypeNameCode,
+    code: Option<TypeCode>,
+}
+
+impl ToTokens for FieldCode {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        if let Some(code) = &self.code {
+            code.to_tokens(tokens)
+        }
+    }
+}
+
+enum TypeCode {
```

Comment:
For all these "code" types, I'd really appreciate a doc comment that points out what they are:

```rust
/// A struct field
///
/// ```rust
/// struct My Struct {
///   my_field: u32 // <-- refers to this
/// }
/// ```
struct FieldCode {
```